###

# Analisis Exploratorio de Datos

Se realiza un analisis exploratorio de datos, con el fin de generar informacion util para que las autoridades locales puedan tomar medidas para reducir la cantidad de victimas falates en siniestros viales.


## Importaciones
Se importan las librerias que se van a utilizar durante el proyecto.
las funciones usadas estan en el archivo py con el nombre functions

In [2]:
import numpy as np
import pandas as pd
import functions

## Pestaña HECHOS
Se extraen los datos de la pestaña HECHOS del excel homicidios.

In [3]:
hechos = pd.read_excel("..\data\homicidios.xlsx", sheet_name= "HECHOS")
hechos

ID  N_VICTIMAS      FECHA  AAAA  MM  DD      HORA  HH  \
0    2016-0001           1 2016-01-01  2016   1   1  04:00:00   4   
1    2016-0002           1 2016-01-02  2016   1   2  01:15:00   1   
2    2016-0003           1 2016-01-03  2016   1   3  07:00:00   7   
3    2016-0004           1 2016-01-10  2016   1  10  00:00:00   0   
4    2016-0005           1 2016-01-21  2016   1  21  05:20:00   5   
..         ...         ...        ...   ...  ..  ..       ...  ..   
691  2021-0093           1 2021-12-13  2021  12  13  17:10:00  17   
692  2021-0094           1 2021-12-20  2021  12  20  01:10:00   1   
693  2021-0095           1 2021-12-30  2021  12  30  00:43:00   0   
694  2021-0096           1 2021-12-15  2021  12  15  10:30:00  10   
695  2021-0097           1 2021-11-18  2021  11  18  06:10:00   6   

                               LUGAR_DEL_HECHO TIPO_DE_CALLE  ...  Altura  \
0    AV PIEDRA BUENA Y AV FERNANDEZ DE LA CRUZ       AVENIDA  ...     NaN   
1            AV GRAL PAZ  Y AV DE LOS CORRALES      GRAL PAZ  ...     NaN   
2                           AV ENTRE RIOS 2034       AVENIDA  ...  2034.0   
3        AV LARRAZABAL Y GRAL VILLEGAS CONRADO       AVENIDA  ...     NaN   
4    AV SAN JUAN Y PRESIDENTE LUIS SAENZ PEÑA        AVENIDA  ...     NaN   
..                                         ...           ...  ...     ...   
691                          AV. RIESTRA Y MOM       AVENIDA  ...     NaN   
692                    AU DELLEPIANE Y LACARRA     AUTOPISTA  ...     NaN   
693                        AV. GAONA Y TERRADA       AVENIDA  ...     NaN   
694                         AV. EVA PERON 4071       AVENIDA  ...  4071.0   
695                    PADRE CARLOS MUJICA 709         CALLE  ...   709.0   

                                   Cruce  \
0    FERNANDEZ DE LA CRUZ, F., GRAL. AV.   
1                    DE LOS CORRALES AV.   
2                                    NaN   
3               VILLEGAS, CONRADO, GRAL.   
4                SAENZ PE?A, LUIS, PRES.   
..                                   ...   
691                                  MOM   
692                          LACARRA AV.   
693                              TERRADA   
694                                  NaN   
695                                  NaN   

                                 Dirección Normalizada COMUNA  \
0    PIEDRA BUENA AV. y FERNANDEZ DE LA CRUZ, F., G...      8   
1                 PAZ, GRAL. AV. y DE LOS CORRALES AV.      9   
2                                  ENTRE RIOS AV. 2034      1   
3            LARRAZABAL AV. y VILLEGAS, CONRADO, GRAL.      8   
4              SAN JUAN AV. y SAENZ PEÑA, LUIS, PRES.      1   
..                                                 ...    ...   
691                                  RIESTRA AV. y MOM      7   
692         DELLEPIANE, LUIS, TTE. GRAL. y LACARRA AV.      9   
693                                GAONA AV. y TERRADA     11   
694                                PERON, EVA AV. 4071      9   
695                            PADRE CARLOS MUGICA 709      1   

                                   XY (CABA)         pos x         pos y  \
0      Point (98896.78238426 93532.43437792)  -58.47533969  -34.68757022   
1      Point (95832.05571093 95505.41641999)  -58.50877521  -34.66977709   
2     Point (106684.29090040 99706.57687843)  -58.39040293  -34.63189362   
3      Point (99840.65224780 94269.16534422)  -58.46503904  -34.68092974   
4    Point (106980.32827929 100752.16915795)  -58.38718297  -34.62246630   
..                                       ...           ...           ...   
691   Point (102728.60090138 98186.24929177)  -58.43353773  -34.64561636   
692    Point (99624.29795829 97569.69801131)  -58.46739825  -34.65117757   
693   Point (99116.45492358 101045.23284826)  -58.47293407  -34.61984745   
694    Point (99324.54463985 97676.26932409)  -58.47066794  -34.65021673   
695  Point (107664.16647795 104708.63962087)  -58.37976155  -34.58679619   

      PARTICIPANTES    VICTIMA    A

Se observan 21 columnas y 696 regritros

Se estandariza el formato de las columnas

In [4]:
#Primer letra en mayuscula.
hechos.columns = [x.capitalize() for x in hechos.columns]
#se remplazan los guoines por espacios
hechos.columns = hechos.columns.str.replace ("_", " ")
#Se remplazan los nombres de algunas columnas para una lectura mas facil.
hechos.rename(columns = {"Id" : "ID",
                            "N victimas" : "Numero Vicitmas",
                            "Aaaa" : "Año",
                            "Mm" : "Mes",
                            "Dd" : "Dia",
                            "Hh" : "Hora Redondeada",
                            "Xy (caba)": "XY (CABA)"}, inplace= True)

In [5]:
hechos.columns

Index(['ID', 'Numero Vicitmas', 'Fecha', 'Año', 'Mes', 'Dia', 'Hora',
       'Hora Redondeada', 'Lugar del hecho', 'Tipo de calle', 'Calle',
       'Altura', 'Cruce', 'Dirección normalizada', 'Comuna', 'XY (CABA)',
       'Pos x', 'Pos y', 'Participantes', 'Victima', 'Acusado'],
      dtype='object')

In [6]:
functions.tipo_dato(hechos)

nombre_campo                                          tipo_dato  \
0                      ID                                    [<class 'str'>]   
1         Numero Vicitmas                                    [<class 'int'>]   
2                   Fecha  [<class 'pandas._libs.tslibs.timestamps.Timest...   
3                     Año                                    [<class 'int'>]   
4                     Mes                                    [<class 'int'>]   
5                     Dia                                    [<class 'int'>]   
6                    Hora  [<class 'datetime.time'>, <class 'str'>, <clas...   
7         Hora Redondeada                     [<class 'int'>, <class 'str'>]   
8         Lugar del hecho                                    [<class 'str'>]   
9           Tipo de calle                                    [<class 'str'>]   
10                  Calle                   [<class 'str'>, <class 'float'>]   
11                 Altura                                  [<class 'float'>]   
12                  Cruce                   [<class 'str'>, <class 'float'>]   
13  Dirección normalizada                   [<class 'str'>, <class 'float'>]   
14                 Comuna                                    [<class 'int'>]   
15              XY (CABA)                                    [<class 'str'>]   
16                  Pos x                                    [<class 'str'>]   
17                  Pos y                                    [<class 'str'>]   
18          Participantes                                    [<class 'str'>]   
19                Victima                                    [<class 'str'>]   
20                Acusado                                    [<class 'str'>]   

    nulo  no_nulo%  nulo%  
0      0    100.00   0.00  
1      0    100.00   0.00  
2      0    100.00   0.00  
3      0    100.00   0.00  
4      0    100.00   0.00  
5      0    100.00   0.00  
6      0    100.00   0.00  
7      0    100.00   0.00  
8      0    100.00   0.00  
9      0    100.00   0.00  
10     1     99.86   0.14  
11   567     18.53  81.47  
12   171     75.43  24.57  
13     8     98.85   1.15  
14     0    100.00   0.00  
15     0    100.00   0.00  
16     0    100.00   0.00  
17     0    100.00   0.00  
18     0    100.00   0.00  
19     0    100.00   0.00  
20     0    100.00   0.00

## Nulos

Se observan varios datos faltantes en  Altura (81.47%), esto sucede por que la mayoria de los siniestros ocurren en esquinas.
Se considera que no es una dato relevante para el analisis, ya que esta misma informacion se encuentra enlas columnas "Lugar del hecho" y "Direccion normalizada".



In [7]:
hechos[["Altura","Lugar del hecho", "Dirección normalizada"]][:3]

Altura                            Lugar del hecho  \
0     NaN  AV PIEDRA BUENA Y AV FERNANDEZ DE LA CRUZ   
1     NaN          AV GRAL PAZ  Y AV DE LOS CORRALES   
2  2034.0                         AV ENTRE RIOS 2034   

                               Dirección normalizada  
0  PIEDRA BUENA AV. y FERNANDEZ DE LA CRUZ, F., G...  
1               PAZ, GRAL. AV. y DE LOS CORRALES AV.  
2                                ENTRE RIOS AV. 2034

Se decide eliminar la columna

In [8]:
hechos.drop(columns = "Altura", inplace= True)

La columna "Cruce" tiene un 24.57% de valores faltantes. Esto sucede cuando el siniestro no ocurrio es una esquina. Este dato se puede utilzar para saber si el hecho ocurrio o no en esquinas.


In [9]:
hechos[hechos["Cruce"].isnull()][["Cruce","Lugar del hecho", "Tipo de calle"]]

Cruce                                    Lugar del hecho Tipo de calle
2     NaN                                 AV ENTRE RIOS 2034       AVENIDA
9     NaN                                 AV ENTRE RIOS 1366       AVENIDA
14    NaN                                       SUIPACHA 156         CALLE
33    NaN                                          LIMA 1483         CALLE
35    NaN  AUTOPISTA 1 SUR PRESIDENTE ARTURO FRONDIZI KM....     AUTOPISTA
..    ...                                                ...           ...
680   NaN                   AU 25 DE MAYO KM7 (ALTURA PASCO)     AUTOPISTA
685   NaN                                     ESMERALDA 1359         CALLE
687   NaN                                     AV. GAONA 3655       AVENIDA
694   NaN                                 AV. EVA PERON 4071       AVENIDA
695   NaN                            PADRE CARLOS MUJICA 709         CALLE

[171 rows x 3 columns]

In [10]:
hechos[hechos["Cruce"].notnull()][["Cruce","Lugar del hecho", "Tipo de calle"]]

Cruce  \
0    FERNANDEZ DE LA CRUZ, F., GRAL. AV.   
1                    DE LOS CORRALES AV.   
3               VILLEGAS, CONRADO, GRAL.   
4                SAENZ PE?A, LUIS, PRES.   
5                           ESCALADA AV.   
..                                   ...   
689                         HUGO, VICTOR   
690                       PUEYRREDON AV.   
691                                  MOM   
692                          LACARRA AV.   
693                              TERRADA   

                               Lugar del hecho Tipo de calle  
0    AV PIEDRA BUENA Y AV FERNANDEZ DE LA CRUZ       AVENIDA  
1            AV GRAL PAZ  Y AV DE LOS CORRALES      GRAL PAZ  
3        AV LARRAZABAL Y GRAL VILLEGAS CONRADO       AVENIDA  
4    AV SAN JUAN Y PRESIDENTE LUIS SAENZ PEÑA        AVENIDA  
5               AV 27 DE FEBRERO Y AV ESCALADA       AVENIDA  
..                                         ...           ...  
689                    BAIGORRIA Y VICTOR HUGO         CALLE  
690             AV. RIVADAVIA Y AV. PUEYRREDON       AVENIDA  
691                          AV. RIESTRA Y MOM       AVENIDA  
692                    AU DELLEPIANE Y LACARRA     AUTOPISTA  
693                        AV. GAONA Y TERRADA       AVENIDA  

[525 rows x 3 columns]

Los datos faltantes son por que los siniestros ocurrieron en algun punto de la calle que no es un cruce.
Se modifica la columna para colocar los Valores "Si" y "No" por si el siniestro ocurrio o no en un cruce. Esto para facilitar la tarea a la hora de graficar.

In [11]:
#Se remplazan los valores nulos por "No" y los no Nulos por "Si"
hechos["Cruce"] = np.where(hechos["Cruce"].notnull(),"SI", "NO")

In [12]:
#Se revisa la columna
hechos[["Cruce", "Lugar del hecho", "Tipo de calle"]]

Cruce                            Lugar del hecho Tipo de calle
0      SI  AV PIEDRA BUENA Y AV FERNANDEZ DE LA CRUZ       AVENIDA
1      SI          AV GRAL PAZ  Y AV DE LOS CORRALES      GRAL PAZ
2      NO                         AV ENTRE RIOS 2034       AVENIDA
3      SI      AV LARRAZABAL Y GRAL VILLEGAS CONRADO       AVENIDA
4      SI  AV SAN JUAN Y PRESIDENTE LUIS SAENZ PEÑA        AVENIDA
..    ...                                        ...           ...
691    SI                          AV. RIESTRA Y MOM       AVENIDA
692    SI                    AU DELLEPIANE Y LACARRA     AUTOPISTA
693    SI                        AV. GAONA Y TERRADA       AVENIDA
694    NO                         AV. EVA PERON 4071       AVENIDA
695    NO                    PADRE CARLOS MUJICA 709         CALLE

[696 rows x 3 columns]

Otra columna con valores faltantes es "Direccion normalizada" la cual tiene solo 8 datis faltantes (1.15%)

In [13]:
hechos[hechos["Dirección normalizada"].isnull()][["Dirección normalizada","Lugar del hecho"]]

Dirección normalizada                          Lugar del hecho
38                    NaN               AUTOPISTA LUGONES PK 10000
106                   NaN         AU BUENOS AIRES - LA PLATA KM. 4
119                   NaN                                       SD
180                   NaN  AU PERITO MORENO Y RAMAL ENLACE AU1/AU6
181                   NaN                       AU DELLEPIANE 2400
313                   NaN                 AUTOPISTA LUGONES KM 4.7
546                   NaN             LUGONES, LEOPOLDO AV. KM 6,1
621                   NaN         AU BUENOS AIRES LA PLATA  KM 4,5

Se observan que no hay informacion que se pueda usar para completar los datos faltantes, pero siguen siendo registros que aportan informacion util, por lo que se decide no eliminarlos, y remplazar los nulos por "SD" (Sin Dato).

In [14]:
hechos["Dirección normalizada"].fillna("SD", inplace= True)

Hay un valor faltante en "Calle"

In [15]:
hechos[hechos["Calle"].isnull()]

ID  Numero Vicitmas      Fecha   Año  Mes  Dia      Hora  \
119  2016-0151                1 2016-11-18  2016   11   18  20:35:00   

    Hora Redondeada Lugar del hecho Tipo de calle Calle Cruce  \
119              20              SD         CALLE   NaN    NO   

    Dirección normalizada  Comuna    XY (CABA) Pos x Pos y Participantes  \
119                    SD       0  Point (. .)     .     .     PEATON-SD   

    Victima Acusado  
119  PEATON      SD

Se observa que no hay datos suficientes como para llenar la informacion. Se remplaza por SD (Sin Dato)

¿Se borra o se deja?

In [16]:
hechos["Calle"].fillna("SD",inplace = True)

## Duplicados

In [17]:
functions.duplicados(hechos, "ID")

'No se encontraron duplicados'

No se encuentran datos duplicados en la columna "ID"

## Variables

In [18]:
functions.tipo_dato(hechos)

nombre_campo                                          tipo_dato  \
0                      ID                                    [<class 'str'>]   
1         Numero Vicitmas                                    [<class 'int'>]   
2                   Fecha  [<class 'pandas._libs.tslibs.timestamps.Timest...   
3                     Año                                    [<class 'int'>]   
4                     Mes                                    [<class 'int'>]   
5                     Dia                                    [<class 'int'>]   
6                    Hora  [<class 'datetime.time'>, <class 'str'>, <clas...   
7         Hora Redondeada                     [<class 'int'>, <class 'str'>]   
8         Lugar del hecho                                    [<class 'str'>]   
9           Tipo de calle                                    [<class 'str'>]   
10                  Calle                                    [<class 'str'>]   
11                  Cruce                                    [<class 'str'>]   
12  Dirección normalizada                                    [<class 'str'>]   
13                 Comuna                                    [<class 'int'>]   
14              XY (CABA)                                    [<class 'str'>]   
15                  Pos x                                    [<class 'str'>]   
16                  Pos y                                    [<class 'str'>]   
17          Participantes                                    [<class 'str'>]   
18                Victima                                    [<class 'str'>]   
19                Acusado                                    [<class 'str'>]   

    nulo  no_nulo%  nulo%  
0      0     100.0    0.0  
1      0     100.0    0.0  
2      0     100.0    0.0  
3      0     100.0    0.0  
4      0     100.0    0.0  
5      0     100.0    0.0  
6      0     100.0    0.0  
7      0     100.0    0.0  
8      0     100.0    0.0  
9      0     100.0    0.0  
10     0     100.0    0.0  
11     0     100.0    0.0  
12     0     100.0    0.0  
13     0     100.0    0.0  
14     0     100.0    0.0  
15     0     100.0    0.0  
16     0     100.0    0.0  
17     0     100.0    0.0  
18     0     100.0    0.0  
19     0     100.0    0.0

Las columnas "Hora" y "Hora Redondeada" tienen varios tipos de datos en la misma columna, por lo que se reisa cada uno.

In [19]:
#Cantidad de valores por tipo
hechos["Hora"].apply(type).value_counts()

Hora
<class 'datetime.time'>        608
<class 'str'>                   85
<class 'datetime.datetime'>      3
Name: count, dtype: int64

Ya que la mayoria de los datos son de tipo datetime.time, se decide convertir todos los datos a este tipo.

In [20]:
hechos["Hora"]  = hechos["Hora"].apply(lambda x: functions.time(x))

In [21]:
#Se verifican los tipos de datos.
hechos["Hora"].apply(type).value_counts()

Hora
<class 'datetime.time'>    695
<class 'NoneType'>           1
Name: count, dtype: int64

In [22]:
hechos[hechos["Hora"].isna()]

ID  Numero Vicitmas      Fecha   Año  Mes  Dia  Hora  \
518  2019-0103                1 2019-12-18  2019   12   18  None   

    Hora Redondeada          Lugar del hecho Tipo de calle           Calle  \
518              SD  PAZ, GRAL. AV. Y GRIVEO      GRAL PAZ  PAZ, GRAL. AV.   

    Cruce    Dirección normalizada  Comuna  \
518    SI  PAZ, GRAL. AV. y GRIVEO      11   

                                  XY (CABA)         Pos x         Pos y  \
518  Point (94643.11254058 103831.57115061)  -58.52169422  -34.59471640   

    Participantes Victima Acusado  
518     MOTO-MOTO    MOTO    MOTO

Falta tanto el dato de "Hora" como el de "Hora Redondeada".
Se decide remplazar por la hora mas comun que suceden homicidios en siniestros viales.

In [23]:
#Se calcula el valor mas comun.
moda = hechos["Hora"].mode().iloc[0]
#Se remplaza el valor None por la hora mas comun.
hechos["Hora"].fillna(moda, inplace= True)
#se verifica el tipo de dato.
hechos["Hora"].apply(type).value_counts()

Hora
<class 'datetime.time'>    696
Name: count, dtype: int64

Ahora se revisa los tipos de dato de "Hora Redondeada"

In [24]:
hechos["Hora Redondeada"].apply(type).value_counts()

Hora Redondeada
<class 'int'>    695
<class 'str'>      1
Name: count, dtype: int64

Se trata de revisa cual es el dato str.
Se ve que es el mismo registro, por que se remplaza por la hora de moda.

In [25]:
hechos.loc[hechos["Hora Redondeada"].apply(lambda x: isinstance(x, str))]

ID  Numero Vicitmas      Fecha   Año  Mes  Dia      Hora  \
518  2019-0103                1 2019-12-18  2019   12   18  09:00:00   

    Hora Redondeada          Lugar del hecho Tipo de calle           Calle  \
518              SD  PAZ, GRAL. AV. Y GRIVEO      GRAL PAZ  PAZ, GRAL. AV.   

    Cruce    Dirección normalizada  Comuna  \
518    SI  PAZ, GRAL. AV. y GRIVEO      11   

                                  XY (CABA)         Pos x         Pos y  \
518  Point (94643.11254058 103831.57115061)  -58.52169422  -34.59471640   

    Participantes Victima Acusado  
518     MOTO-MOTO    MOTO    MOTO

In [26]:
#Se remplaza el valor None por la hora mas comun.
hechos["Hora Redondeada"] = hechos["Hora Redondeada"].apply(lambda x: int(moda.hour) if x == "SD" else x)

#se verifica el tipo de dato.
hechos["Hora Redondeada"].apply(type).value_counts()


Hora Redondeada
<class 'int'>    696
Name: count, dtype: int64

## Victima

Se revisan las categorias de victimas.

In [27]:
hechos["Victima"].unique()

array(['MOTO', 'AUTO', 'PEATON', 'SD', 'CARGAS', 'BICICLETA', 'PASAJEROS',
       'MOVIL', 'OBJETO FIJO', 'PEATON_MOTO'], dtype=object)

Se observan las categortias "PEATON_MOTO" y "OBJETO FIJO" que no estan en el diccionario de datos.

In [28]:
#Se observan los registros anomalos para corregirlos.
hechos[hechos["Victima"].isin(["OBJETO FIJO","PEATON_MOTO"])]

ID  Numero Vicitmas      Fecha   Año  Mes  Dia      Hora  \
230  2017-0108                2 2017-09-02  2017    9    2  04:53:08   
583  2020-0063                2 2020-12-05  2020   12    5  07:10:00   

     Hora Redondeada           Lugar del hecho Tipo de calle           Calle  \
230                4   AV. GRAL. PAZ Y MACHAIN      GRAL PAZ  PAZ, GRAL. AV.   
583                7  NUEVA YORK Y ALTA GRACIA         CALLE      NUEVA YORK   

    Cruce     Dirección normalizada  Comuna  \
230    SI  PAZ, GRAL. AV. y MACHAIN      12   
583    SI  NUEVA YORK y ALTA GRACIA      11   

                                  XY (CABA)         Pos x         Pos y  \
230  Point (97098.48468623 109019.96106626)  -58.49491054  -34.54795581   
583  Point (94080.62190808 102083.62453795)  -58.52783814  -34.61047001   

        Participantes      Victima Acusado  
230  AUTO-OBJETO FIJO  OBJETO FIJO    AUTO  
583  PEATON_MOTO-MOTO  PEATON_MOTO    MOTO

Son un registro de cada tipo. Se transforman en "OTRO" ya que no hya datos que indiquen el tipo de victima.

In [29]:
#Se remplaza "OBJETO FIJO" por "OTRO"
hechos["Victima"].replace("OBJETO FIJO","OTRO",inplace=True)
#Se remplaza "PEATON_MOTO" por "OTRO"
hechos["Victima"].replace("PEATON_MOTO","OTRO",inplace=True)

## Acusado
Se revisa la columna Acusado

In [30]:
hechos["Acusado"].unique()

array(['AUTO', 'PASAJEROS', 'SD', 'OBJETO FIJO', 'CARGAS', 'MOTO',
       'MULTIPLE', 'OTRO', 'BICICLETA', 'TREN'], dtype=object)

No se observan caracteristicas fuera de las esperadas.

## Coordenadas Geograficas
Se obserban faltantes de las columnas "XY (CABA)" , "Pos x" y "Pos y"

In [31]:
hechos[hechos["XY (CABA)"] == "Point (. .)"]

ID  Numero Vicitmas      Fecha   Año  Mes  Dia      Hora  \
35   2016-0049                1 2016-04-17  2016    4   17  00:00:00   
38   2016-0052                1 2016-04-20  2016    4   20  20:00:00   
71   2016-0096                1 2016-07-25  2016    7   25  07:00:00   
106  2016-0136                1 2016-10-25  2016   10   25  00:00:00   
119  2016-0151                1 2016-11-18  2016   11   18  20:35:00   
139  2016-0174                1 2016-12-27  2016   12   27  00:00:00   
176  2017-0042                1 2017-04-10  2017    4   10  09:00:00   
180  2017-0050                2 2017-04-28  2017    4   28  11:08:08   
181  2017-0051                1 2017-05-01  2017    5    1  03:47:47   
256  2017-0140                1 2017-11-19  2017   11   19  23:22:17   
313  2018-0039                1 2018-04-21  2018    4   21  22:15:00   
546  2020-0026                1 2020-05-17  2020    5   17  06:40:00   
559  2020-0039                1 2020-09-01  2020    9    1  19:17:42   
621  2021-0023                1 2021-03-01  2021    3    1  09:20:00   

     Hora Redondeada                                    Lugar del hecho  \
35                 0  AUTOPISTA 1 SUR PRESIDENTE ARTURO FRONDIZI KM....   
38                20                         AUTOPISTA LUGONES PK 10000   
71                 7       AUTOPISTA DELLEPIANE LUIS TTE. GRAL. KM. 2,3   
106                0                   AU BUENOS AIRES - LA PLATA KM. 4   
119               20                                                 SD   
139                0                               AUTOPISTA 25 DE MAYO   
176                9                      AV. LEOPOLDO LUGONES PKM 6900   
180               11            AU PERITO MORENO Y RAMAL ENLACE AU1/AU6   
181                3                                 AU DELLEPIANE 2400   
256               23                        AU ARTURO FRONDIZI PKM 3100   
313               22                           AUTOPISTA LUGONES KM 4.7   
546                6                       LUGONES, LEOPOLDO AV. KM 6,1   
559               19                                    MURGUIONDO 2700   
621                9                   AU BUENOS AIRES LA PLATA  KM 4,5   

    Tipo de calle                                       Calle Cruce  \
35      AUTOPISTA  AUTOPISTA 1 SUR PRESIDENTE ARTURO FRONDIZI    NO   
38      AUTOPISTA                       LUGONES, LEOPOLDO AV.    NO   
71      AUTOPISTA        AUTOPISTA DELLEPIANE LUIS TTE. GRAL.    NO   
106     AUTOPISTA           AUTOPISTA BUENOS AIRES - LA PLATA    NO   
119         CALLE                                          SD    NO   
139     AUTOPISTA                        AUTOPISTA 25 DE MAYO    NO   
176      GRAL PAZ                       LUGONES, LEOPOLDO AV.    NO   
180     AUTOPISTA                     AUTOPISTA PERITO MORENO    NO   
181     AUTOPISTA        AUTOPISTA DELLEPIANE LUIS TTE. GRAL.    NO   
256     AUTOPISTA  AUTOPISTA 1 SUR PRESIDENTE ARTURO FRONDIZI    NO   
313     AUTOPISTA                       LUGONES, LEOPOLDO AV.    NO   
546     AUTOPISTA                       LUGONES, LEOPOLDO AV.    NO   
559         CALLE                                  MURGUIONDO    NO   
621     AUTOPISTA           AUTOPISTA BUENOS AIRES - LA PLATA    NO   

                          Dirección normalizada  Comuna    XY (CABA)  \
35   AUTOPISTA 1 SUR PRESIDENTE ARTURO FRONDIZI       4  Point (. .)   
38                                           SD      13  Point (. .)   
71         AUTOPISTA DELLEPIANE LUIS TTE. GRAL.       8  Point (. .)   
106                                          SD       4  Point (. .)   
119                                          SD       0  Point (. .)   
139                        AUTOPISTA 25 DE MAYO       0  Point (. .)   
176                       LUGONES, LEOPOLDO AV.      14  Point (. .)   
180                                          SD       9  Point (. .)   
181                                          SD       7  Point (. .)   
256  AUTOPISTA 1 SU

Hay 14 registros que no contienen coordenadas planas, se trato de conseguir informacion para obtener dichas coordenadas, pero hasta el momento no se pudo hacer.
Tambien hay 12 registros que no presentan informacion de "Pos x" y "Pos y".
Se deicidio remplazar los valores faltantes por 0 de tal manera que se pueda diferenciar en el dashboard.

In [32]:
hechos["XY (CABA)"] = hechos["XY (CABA)"].replace("Point (. .)", 0)
hechos["Pos x"] = hechos["Pos x"].replace(".", 0)
hechos["Pos y"] = hechos["Pos y"].replace(".", 0)

## Hoja VICTIMAS

Se extraen los datos de la hoja "VICTIMAS" del excel homicidios

In [33]:
victimas = pd.read_excel("..\data\homicidios.xlsx",sheet_name = "VICTIMAS")
victimas

ID_hecho      FECHA  AAAA  MM  DD                   ROL VICTIMA  \
0    2016-0001 2016-01-01  2016   1   1             CONDUCTOR    MOTO   
1    2016-0002 2016-01-02  2016   1   2             CONDUCTOR    AUTO   
2    2016-0003 2016-01-03  2016   1   3             CONDUCTOR    MOTO   
3    2016-0004 2016-01-10  2016   1  10             CONDUCTOR    MOTO   
4    2016-0005 2016-01-21  2016   1  21             CONDUCTOR    MOTO   
..         ...        ...   ...  ..  ..                   ...     ...   
712  2021-0092 2021-12-12  2021  12  12                PEATON  PEATON   
713  2021-0093 2021-12-13  2021  12  13  PASAJERO_ACOMPAÑANTE    MOTO   
714  2021-0094 2021-12-20  2021  12  20  PASAJERO_ACOMPAÑANTE    MOTO   
715  2021-0095 2021-12-30  2021  12  30             CONDUCTOR    MOTO   
716  2021-0096 2021-12-15  2021  12  15             CONDUCTOR    AUTO   

          SEXO EDAD  FECHA_FALLECIMIENTO  
0    MASCULINO   19  2016-01-01 00:00:00  
1    MASCULINO   70  2016-01-02 00:00:00  
2    MASCULINO   30  2016-01-03 00:00:00  
3    MASCULINO   18                   SD  
4    MASCULINO   29  2016-02-01 00:00:00  
..         ...  ...                  ...  
712   FEMENINO   50  2021-12-12 00:00:00  
713   FEMENINO   18  2021-12-18 00:00:00  
714   FEMENINO   43  2021-12-20 00:00:00  
715  MASCULINO   27  2022-01-02 00:00:00  
716  MASCULINO   60  2021-12-20 00:00:00  

[717 rows x 10 columns]

Se normalizan los nombres de las columnas

In [34]:
#Primer letra en mayuscula.
victimas.columns = [x.capitalize() for x in victimas.columns]
#se remplazan los guoines por espacios
victimas.columns = victimas.columns.str.replace ("_", " ")
#Se remplazan los nombres de algunas columnas para una lectura mas facil.
victimas.rename(columns = {"Id hecho" : "ID",
                            "Aaaa" : "Año",
                            "Mm" : "Mes",
                            "Dd" : "Dia"}, inplace= True)

In [35]:
victimas

ID      Fecha   Año  Mes  Dia                   Rol Victima  \
0    2016-0001 2016-01-01  2016    1    1             CONDUCTOR    MOTO   
1    2016-0002 2016-01-02  2016    1    2             CONDUCTOR    AUTO   
2    2016-0003 2016-01-03  2016    1    3             CONDUCTOR    MOTO   
3    2016-0004 2016-01-10  2016    1   10             CONDUCTOR    MOTO   
4    2016-0005 2016-01-21  2016    1   21             CONDUCTOR    MOTO   
..         ...        ...   ...  ...  ...                   ...     ...   
712  2021-0092 2021-12-12  2021   12   12                PEATON  PEATON   
713  2021-0093 2021-12-13  2021   12   13  PASAJERO_ACOMPAÑANTE    MOTO   
714  2021-0094 2021-12-20  2021   12   20  PASAJERO_ACOMPAÑANTE    MOTO   
715  2021-0095 2021-12-30  2021   12   30             CONDUCTOR    MOTO   
716  2021-0096 2021-12-15  2021   12   15             CONDUCTOR    AUTO   

          Sexo Edad  Fecha fallecimiento  
0    MASCULINO   19  2016-01-01 00:00:00  
1    MASCULINO   70  2016-01-02 00:00:00  
2    MASCULINO   30  2016-01-03 00:00:00  
3    MASCULINO   18                   SD  
4    MASCULINO   29  2016-02-01 00:00:00  
..         ...  ...                  ...  
712   FEMENINO   50  2021-12-12 00:00:00  
713   FEMENINO   18  2021-12-18 00:00:00  
714   FEMENINO   43  2021-12-20 00:00:00  
715  MASCULINO   27  2022-01-02 00:00:00  
716  MASCULINO   60  2021-12-20 00:00:00  

[717 rows x 10 columns]

### Nulos
Se revisa que no hay datos nulos.

In [36]:
victimas.isnull().any()

ID                     False
Fecha                  False
Año                    False
Mes                    False
Dia                    False
Rol                    False
Victima                False
Sexo                   False
Edad                   False
Fecha fallecimiento    False
dtype: bool

### Duplicados

Se verifica si hay ID duplicados

In [37]:
functions.duplicados(victimas,"ID")

ID      Fecha   Año  Mes  Dia                   Rol Victima  \
29   2016-0041 2016-03-29  2016    3   29             CONDUCTOR    MOTO   
30   2016-0041 2016-03-29  2016    3   29  PASAJERO_ACOMPAÑANTE    MOTO   
98   2016-0126 2016-09-18  2016    9   18             CONDUCTOR    AUTO   
99   2016-0126 2016-09-18  2016    9   18  PASAJERO_ACOMPAÑANTE    AUTO   
163  2017-0026 2017-02-26  2017    2   26  PASAJERO_ACOMPAÑANTE    AUTO   
164  2017-0026 2017-02-26  2017    2   26             CONDUCTOR    AUTO   
173  2017-0035 2017-03-23  2017    3   23             CONDUCTOR    AUTO   
174  2017-0035 2017-03-23  2017    3   23  PASAJERO_ACOMPAÑANTE    AUTO   
175  2017-0035 2017-03-23  2017    3   23  PASAJERO_ACOMPAÑANTE    AUTO   
177  2017-0036 2017-03-29  2017    3   29             CONDUCTOR    MOTO   
176  2017-0036 2017-03-29  2017    3   29  PASAJERO_ACOMPAÑANTE    MOTO   
186  2017-0050 2017-04-28  2017    4   28             CONDUCTOR    MOTO   
187  2017-0050 2017-04-28  2017    4   28  PASAJERO_ACOMPAÑANTE    MOTO   
237  2017-0108 2017-09-02  2017    9    2  PASAJERO_ACOMPAÑANTE    AUTO   
238  2017-0108 2017-09-02  2017    9    2  PASAJERO_ACOMPAÑANTE    AUTO   
242  2017-0112 2017-09-10  2017    9   10  PASAJERO_ACOMPAÑANTE    AUTO   
243  2017-0112 2017-09-10  2017    9   10  PASAJERO_ACOMPAÑANTE    AUTO   
253  2017-0126 2017-10-14  2017   10   14  PASAJERO_ACOMPAÑANTE    AUTO   
254  2017-0126 2017-10-14  2017   10   14             CONDUCTOR    AUTO   
278  2017-0153 2017-12-11  2017   12   11             CONDUCTOR    MOTO   
277  2017-0153 2017-12-11  2017   12   11  PASAJERO_ACOMPAÑANTE    MOTO   
300  2018-0015 2018-02-06  2018    2    6             CONDUCTOR    MOTO   
301  2018-0015 2018-02-06  2018    2    6  PASAJERO_ACOMPAÑANTE    MOTO   
312  2018-0026 2018-03-09  2018    3    9             CONDUCTOR   MOVIL   
313  2018-0026 2018-03-09  2018    3    9  PASAJERO_ACOMPAÑANTE   MOVIL   
325  2018-0038 2018-04-20  2018    4   20                PEATON  PEATON   
326  2018-0038 2018-04-20  2018    4   20                PEATON  PEATON   
373  2018-0085 2018-08-03  2018    8    3  PASAJERO_ACOMPAÑANTE    MOTO   
374  2018-0085 2018-08-03  2018    8    3             CONDUCTOR    MOTO   
388  2018-0098 2018-09-09  2018    9    9  PASAJERO_ACOMPAÑANTE    AUTO   
387  2018-0098 2018-09-09  2018    9    9             CONDUCTOR    AUTO   
418  2018-0128 2018-11-25  2018   11   25  PASAJERO_ACOMPAÑANTE    AUTO   
419  2018-0128 2018-11-25  2018   11   25  PASAJERO_ACOMPAÑANTE    AUTO   
444  2019-0010 2019-02-02  2019    2    2                PEATON  PEATON   
445  2019-0010 2019-02-02  2019    2    2                PEATON  PEATON   
576  2020-0038 2020-08-30  2020    8   30             CONDUCTOR    AUTO   
577  2020-0038 2020-08-30  2020    8   30             CONDUCTOR    AUTO   
602  2020-0063 2020-12-05  2020   12    5                PEATON  PEATON   
603  2020-0063 2020-12-05  2020   12    5  PASAJERO_ACOMPAÑANTE    MOTO   
616  2020-0077 2020-12-25  2020   12   25             CONDUCTOR    MOTO   
617  2020-0077 2020-12-25  2020   12   25  PASAJERO_ACOMPAÑANTE    MOTO   

          Sexo Edad  Fecha fallecimiento  
29   MASCULINO   54  2016-03-29 00:00:00  
30   MASCULINO   SD  2016-03-30 00:00:00  
98   MASCULINO   37                   SD  
99   MASCULINO   60                   SD  
163   FEMENINO   23  2017-02-26 00:00:00  
164  MASCULINO   19  2017-02-26 00:00:00  
173  MASCULINO   28  2017-03-23 00:00:00  
174  MASCULINO   32  2017-03-23 00:00:00  
175  MASCULINO   30  2017-03-23 00:00:00  
177  MASCULINO   20  2017-03-29 00:00:00  
176   FEMENINO   SD  2017-03-29 00:00:00  
186  MASCULINO   46  2017-04-28 00:00:00  
187  MASCULINO   16  2017-04-28 00:00:00  
237  MASCULINO   17  2017-09-02 00:00:00  
238  MASCULINO   10  2017-09-02 00:00:00  
242  MASCULINO    1                   SD  
243  MASCULINO   SD  2017-09-13 00:00:00  
253  MASCULINO   39                   SD  
254  MASCULINO   36  2017-10-14 00:00:00  
2

Se observan varios "ID" repetidos, pero corresponden al mismo siniestro vial, solo que diferentes vitcimas. Por lo tanto se decide dejarlos.

## Nulos

Se verifica si hay datos nulos

In [38]:
functions.tipo_dato(victimas)

nombre_campo                                          tipo_dato  \
0                   ID                                    [<class 'str'>]   
1                Fecha  [<class 'pandas._libs.tslibs.timestamps.Timest...   
2                  Año                                    [<class 'int'>]   
3                  Mes                                    [<class 'int'>]   
4                  Dia                                    [<class 'int'>]   
5                  Rol                                    [<class 'str'>]   
6              Victima                                    [<class 'str'>]   
7                 Sexo                                    [<class 'str'>]   
8                 Edad                     [<class 'int'>, <class 'str'>]   
9  Fecha fallecimiento       [<class 'datetime.datetime'>, <class 'str'>]   

   nulo  no_nulo%  nulo%  
0     0     100.0    0.0  
1     0     100.0    0.0  
2     0     100.0    0.0  
3     0     100.0    0.0  
4     0     100.0    0.0  
5     0     100.0    0.0  
6     0     100.0    0.0  
7     0     100.0    0.0  
8     0     100.0    0.0  
9     0     100.0    0.0

Se observa que la columnas de Edad y fecha de fallecimiento tienen varios tipos de datos. Por lo tanto se revisan para asigrnar un solo tipo de dato

In [39]:
#Se observa primero "Edad".
victimas["Edad"].apply(type).value_counts()


Edad
<class 'int'>    664
<class 'str'>     53
Name: count, dtype: int64

Son 53 datos en str, esto sucede ya que no se tiene el dato de la "Edad", por que se decide remplazarlos por la edad promedio de fallecimiento. Para hacer esto se compara tambien el "Sexo", "Rol" y "Victima". remplazando en caso de que falte por el mas comun.

In [40]:
victimas[victimas["Edad"].apply(lambda x: isinstance (x,str))]

ID      Fecha   Año  Mes  Dia                   Rol    Victima  \
30   2016-0041 2016-03-29  2016    3   29  PASAJERO_ACOMPAÑANTE       MOTO   
33   2016-0045 2016-04-11  2016    4   11             CONDUCTOR       MOTO   
35   2016-0048 2016-04-15  2016    4   15                PEATON     PEATON   
36   2016-0049 2016-04-17  2016    4   17                    SD         SD   
39   2016-0052 2016-04-20  2016    4   20                    SD       MOTO   
55   2016-0077 2016-06-13  2016    6   13                PEATON     PEATON   
63   2016-0085 2016-06-29  2016    6   29                    SD       MOTO   
72   2016-0096 2016-07-25  2016    7   25             CONDUCTOR       MOTO   
89   2016-0115 2016-09-02  2016    9    2                    SD         SD   
93   2016-0119 2016-09-04  2016    9    4  PASAJERO_ACOMPAÑANTE         SD   
108  2016-0136 2016-10-25  2016   10   25             CONDUCTOR       MOTO   
119  2016-0148 2016-11-16  2016   11   16             CONDUCTOR       MOTO   
121  2016-0151 2016-11-18  2016   11   18                PEATON     PEATON   
138  2016-0171 2016-12-25  2016   12   25             CONDUCTOR       MOTO   
141  2016-0174 2016-12-27  2016   12   27                    SD         SD   
142  2016-0175 2016-12-27  2016   12   27                PEATON     PEATON   
165  2017-0027 2017-02-26  2017    2   26                PEATON     PEATON   
176  2017-0036 2017-03-29  2017    3   29  PASAJERO_ACOMPAÑANTE       MOTO   
182  2017-0042 2017-04-10  2017    4   10             CONDUCTOR       MOTO   
225  2017-0093 2017-07-25  2017    7   25                PEATON     PEATON   
243  2017-0112 2017-09-10  2017    9   10  PASAJERO_ACOMPAÑANTE       AUTO   
275  2017-0151 2017-12-04  2017   12    4                PEATON     PEATON   
277  2017-0153 2017-12-11  2017   12   11  PASAJERO_ACOMPAÑANTE       MOTO   
299  2018-0014 2018-01-31  2018    1   31                PEATON     PEATON   
328  2018-0040 2018-04-27  2018    4   27                PEATON     PEATON   
359  2018-0071 2018-07-10  2018    7   10             CONDUCTOR       MOTO   
367  2018-0079 2018-07-23  2018    7   23                PEATON     PEATON   
371  2018-0083 2018-08-02  2018    8    2             CONDUCTOR       MOTO   
377  2018-0088 2018-08-17  2018    8   17             CONDUCTOR       MOTO   
391  2018-0101 2018-09-21  2018    9   21             CONDUCTOR       AUTO   
421  2018-0130 2018-12-03  2018   12    3             CONDUCTOR       MOTO   
444  2019-0010 2019-02-02  2019    2    2                PEATON     PEATON   
446  2019-0011 2019-02-10  2019    2   10                PEATON     PEATON   
454  2019-0019 2019-03-06  2019    3    6             CONDUCTOR       MOTO   
457  2019-0022 2019-03-15  2019    3   15  PASAJERO_ACOMPAÑANTE       AUTO   
461  2019-0026 2019-03-28  2019    3   28                PEATON     PEATON   
463  2019-0028 2019-04-06  2019    4    6                PEATON     PEATON   
469  2019-0034 2019-04-21  2019    4   21             CONDUCTOR       MOTO   
479  2019-0043 2019-05-21  2019    5   21              CICLISTA  BICICLETA   
498  2019-0063 2019-08-04  2019    8    4  PASAJERO_ACOMPAÑANTE       MOTO   
506  2019-0071 2019-08-27  2019    8   27                PEATON     PEATON   
529  2019-0094 2019-11-18  2019   11   18  PASAJERO_ACOMPAÑANTE       MOTO   
530  2019-0095 2019-11-20  2019   11   20             CONDUCTOR       MOTO   
548  2020-0010 2020-02-08  2020    2    8             CONDUCTOR       MOTO   
552  2020-0014 2020-02-27  2020    2   27             CONDUCTOR       MOTO   
555  2020-0017 2020-03-05  2020    3    5                PEATON     PEATON   
563  2020-0025 2020-05-12  2020    5   12                PEATON     PEATON   
572  2020-0034 2020-08-02  2020    8    2                PEATON     PEATON   
579  2020-0040 2020-09-12  2020    9   12             CONDUCTOR       AUTO   
631  2021-0012 2021-01-29  2021    1   29                PEATON     PEATON   
639  2021-0020 2021-02-21  2021   

In [41]:
# Se reemplaza "SD" con NaN en la columna
victimas["Sexo"].replace("SD", pd.NA,inplace= True)
# Se calcula el valor mas comun en la columna
moda = victimas["Sexo"].mode().iloc[0]
print(f"El valor mas comun es {moda}")
#Se remplazan los valores Nulos con el valor mas frecuente.
victimas["Sexo"].fillna(moda,inplace=True)


El valor mas comun es MASCULINO


Una vez que remplazamos el "Sexo" con "MASCULINO" por ser el valor mas comun, se remplaza la edad mas comun segun el sexo.

In [42]:
#Se remplaza "SD" con Nan en la columna
victimas["Edad"].replace("SD", pd.NA, inplace= True)
#Se calcula el promedio de edad segun el "Sexo"
edad_sexo = victimas.groupby("Sexo")["Edad"].mean()
print(f'La edad promedio de Masculino es {round(edad_sexo["MASCULINO"])}')
print(f'La edad promedio de Femenino es {round(edad_sexo["FEMENINO"])}')
#Se remplazan los nulos por el promedio de edad segun el sexo.
victimas["Edad"] = victimas.apply(lambda row: edad_sexo[row["Sexo"]] if pd.isna(row["Edad"]) else row["Edad"],axis =1)
#Se convierte a entero.
victimas["Edad"] = victimas["Edad"].astype(int)

La edad promedio de Masculino es 40
La edad promedio de Femenino es 51


Ahora se revisan las columnas de "Rol" y "Victima"

In [43]:
#Cantidad de datos faltantes en columna "Rol"
len(victimas[victimas["Rol"]== "SD"])

11

In [44]:
#Cantidad de datos faltantes en columna "Victima"
len(victimas[victimas["Victima"]== "SD"])

9

Son 11 y 9 datos que faltan en las columnas de "Rol" y "Victima" respectivamente, se las remplaza por los valores mas comunes.

In [45]:
# Se reemplaza "SD" con NaN en la columna
victimas["Rol"].replace("SD", pd.NA,inplace= True)
# Se calcula el valor mas comun en la columna
moda = victimas["Rol"].mode().iloc[0]
print(f"El valor mas comun es {moda}")
#Se remplazan los valores Nulos con el valor mas frecuente.
victimas["Rol"].fillna(moda,inplace=True)


El valor mas comun es CONDUCTOR


In [46]:
# Se reemplaza "SD" con NaN en la columna
victimas["Victima"].replace("SD", pd.NA,inplace= True)
# Se calcula el valor mas comun en la columna
moda = victimas["Victima"].mode().iloc[0]
print(f"El valor mas comun es {moda}")
#Se remplazan los valores Nulos con el valor mas frecuente.
victimas["Victima"].fillna(moda,inplace=True)


El valor mas comun es MOTO


La columna "Fecha Fallecimiento", no aporta informacion util para el objetivo de reducir la cantidad de fallecimientos en siniestros viales, por lo tanto se decide eliminarla. 

In [47]:
victimas.drop("Fecha fallecimiento", axis = 1, inplace= True)
victimas.head()

ID      Fecha   Año  Mes  Dia        Rol Victima       Sexo  Edad
0  2016-0001 2016-01-01  2016    1    1  CONDUCTOR    MOTO  MASCULINO    19
1  2016-0002 2016-01-02  2016    1    2  CONDUCTOR    AUTO  MASCULINO    70
2  2016-0003 2016-01-03  2016    1    3  CONDUCTOR    MOTO  MASCULINO    30
3  2016-0004 2016-01-10  2016    1   10  CONDUCTOR    MOTO  MASCULINO    18
4  2016-0005 2016-01-21  2016    1   21  CONDUCTOR    MOTO  MASCULINO    29

Se vuelve a verificar todos los tipos de datos.

In [48]:
functions.tipo_dato(victimas)

nombre_campo                                          tipo_dato  nulo  \
0           ID                                    [<class 'str'>]     0   
1        Fecha  [<class 'pandas._libs.tslibs.timestamps.Timest...     0   
2          Año                                    [<class 'int'>]     0   
3          Mes                                    [<class 'int'>]     0   
4          Dia                                    [<class 'int'>]     0   
5          Rol                                    [<class 'str'>]     0   
6      Victima                                    [<class 'str'>]     0   
7         Sexo                                    [<class 'str'>]     0   
8         Edad                                    [<class 'int'>]     0   

   no_nulo%  nulo%  
0     100.0    0.0  
1     100.0    0.0  
2     100.0    0.0  
3     100.0    0.0  
4     100.0    0.0  
5     100.0    0.0  
6     100.0    0.0  
7     100.0    0.0  
8     100.0    0.0

Se eliminan las columnas "Fecha", "Año", "Mes" , "Dia"  y "Victima" por que el dataframe se va a unir al de hechos para hacer los analisis, y este ya tiene esos datos. 

In [49]:
columnas = ["Fecha","Año","Mes","Dia", "Victima"]
victimas.drop(columnas,axis =1, inplace= True)

## Union de los dataframe.
Se unen los dos dataframe en un unico dataframe.

In [50]:
homicidios = hechos.merge(victimas,on = "ID", how = "left")

In [51]:
homicidios

ID  Numero Vicitmas      Fecha   Año  Mes  Dia      Hora  \
0    2016-0001                1 2016-01-01  2016    1    1  04:00:00   
1    2016-0002                1 2016-01-02  2016    1    2  01:15:00   
2    2016-0003                1 2016-01-03  2016    1    3  07:00:00   
3    2016-0004                1 2016-01-10  2016    1   10  00:00:00   
4    2016-0005                1 2016-01-21  2016    1   21  05:20:00   
..         ...              ...        ...   ...  ...  ...       ...   
712  2021-0093                1 2021-12-13  2021   12   13  17:10:00   
713  2021-0094                1 2021-12-20  2021   12   20  01:10:00   
714  2021-0095                1 2021-12-30  2021   12   30  00:43:00   
715  2021-0096                1 2021-12-15  2021   12   15  10:30:00   
716  2021-0097                1 2021-11-18  2021   11   18  06:10:00   

     Hora Redondeada                            Lugar del hecho Tipo de calle  \
0                  4  AV PIEDRA BUENA Y AV FERNANDEZ DE LA CRUZ       AVENIDA   
1                  1          AV GRAL PAZ  Y AV DE LOS CORRALES      GRAL PAZ   
2                  7                         AV ENTRE RIOS 2034       AVENIDA   
3                  0      AV LARRAZABAL Y GRAL VILLEGAS CONRADO       AVENIDA   
4                  5  AV SAN JUAN Y PRESIDENTE LUIS SAENZ PEÑA        AVENIDA   
..               ...                                        ...           ...   
712               17                          AV. RIESTRA Y MOM       AVENIDA   
713                1                    AU DELLEPIANE Y LACARRA     AUTOPISTA   
714                0                        AV. GAONA Y TERRADA       AVENIDA   
715               10                         AV. EVA PERON 4071       AVENIDA   
716                6                    PADRE CARLOS MUJICA 709         CALLE   

     ... Comuna                                XY (CABA)         Pos x  \
0    ...      8    Point (98896.78238426 93532.43437792)  -58.47533969   
1    ...      9    Point (95832.05571093 95505.41641999)  -58.50877521   
2    ...      1   Point (106684.29090040 99706.57687843)  -58.39040293   
3    ...      8    Point (99840.65224780 94269.16534422)  -58.46503904   
4    ...      1  Point (106980.32827929 100752.16915795)  -58.38718297   
..   ...    ...                                      ...           ...   
712  ...      7   Point (102728.60090138 98186.24929177)  -58.43353773   
713  ...      9    Point (99624.29795829 97569.69801131)  -58.46739825   
714  ...     11   Point (99116.45492358 101045.23284826)  -58.47293407   
715  ...      9    Point (99324.54463985 97676.26932409)  -58.47066794   
716  ...      1  Point (107664.16647795 104708.63962087)  -58.37976155   

            Pos y   Participantes    Victima    Acusado                   Rol  \
0    -34.68757022       MOTO-AUTO       MOTO       AUTO             CONDUCTOR   
1    -34.66977709  AUTO-PASAJEROS       AUTO  PASAJEROS             CONDUCTOR   
2    -34.63189362       MOTO-AUTO       MOTO       AUTO             CONDUCTOR   
3    -34.68092974         MOTO-SD       MOTO         SD             CONDUCTOR   
4    -34.62246630  MOTO-PASAJEROS       MOTO  PASAJEROS             CONDUCTOR   
..            ...             ...        ...        ...                   ...   
712  -34.64561636       MOTO-AUTO       MOTO       AUTO  PASAJERO_ACOMPAÑANTE   
713  -34.65117757       MOTO-AUTO       MOTO       AUTO  PASAJERO_ACOMPAÑANTE   
714  -34.61984745     MOTO-CARGAS       MOTO     CARGAS             CONDUCTOR   
715  -34.65021673     AUTO-CARGAS       AUTO     CARGAS             CONDUCTOR   
716  -34.58679619  BICICLETA-AUTO  BICICLETA       AUTO              CICLISTA   

          Sexo Edad  
0    MASCULINO   19  
1    MASCULINO   70  
2    MASCULINO   30  
3    MASCULINO   18  
4    MASCULINO   29  
..         ...  ...  
712   FEMENINO   18  
713   FEMENINO   43  
714  MASCULINO   27  
715  MASCULINO   60  
716  MASCULINO   53  

[717 rows x 23 columns]

In [52]:
#Se revisan los datos una vez mas.
functions.tipo_dato(homicidios)

nombre_campo                                          tipo_dato  \
0                      ID                                    [<class 'str'>]   
1         Numero Vicitmas                                    [<class 'int'>]   
2                   Fecha  [<class 'pandas._libs.tslibs.timestamps.Timest...   
3                     Año                                    [<class 'int'>]   
4                     Mes                                    [<class 'int'>]   
5                     Dia                                    [<class 'int'>]   
6                    Hora                          [<class 'datetime.time'>]   
7         Hora Redondeada                                    [<class 'int'>]   
8         Lugar del hecho                                    [<class 'str'>]   
9           Tipo de calle                                    [<class 'str'>]   
10                  Calle                                    [<class 'str'>]   
11                  Cruce                                    [<class 'str'>]   
12  Dirección normalizada                                    [<class 'str'>]   
13                 Comuna                                    [<class 'int'>]   
14              XY (CABA)                     [<class 'str'>, <class 'int'>]   
15                  Pos x                     [<class 'str'>, <class 'int'>]   
16                  Pos y                     [<class 'str'>, <class 'int'>]   
17          Participantes                                    [<class 'str'>]   
18                Victima                                    [<class 'str'>]   
19                Acusado                                    [<class 'str'>]   
20                    Rol                                    [<class 'str'>]   
21                   Sexo                                    [<class 'str'>]   
22                   Edad                                    [<class 'int'>]   

    nulo  no_nulo%  nulo%  
0      0     100.0    0.0  
1      0     100.0    0.0  
2      0     100.0    0.0  
3      0     100.0    0.0  
4      0     100.0    0.0  
5      0     100.0    0.0  
6      0     100.0    0.0  
7      0     100.0    0.0  
8      0     100.0    0.0  
9      0     100.0    0.0  
10     0     100.0    0.0  
11     0     100.0    0.0  
12     0     100.0    0.0  
13     0     100.0    0.0  
14     0     100.0    0.0  
15     0     100.0    0.0  
16     0     100.0    0.0  
17     0     100.0    0.0  
18     0     100.0    0.0  
19     0     100.0    0.0  
20     0     100.0    0.0  
21     0     100.0    0.0  
22     0     100.0    0.0

No se ven valores nulos, y las columnas tienen un solo tipo de dato. Se puede pasar al Analisis Exploratorio de los datos.

In [53]:
#Se guarda el dataframe en un csv
homicidios.to_csv("..\data\homicidios.csv")